specifications/todo-api.yaml

```diff
@@ -0,0 +1,99 @@
+openapi: 3.0.2
+
+info:
+  title: TODO API
+  description: |-
+    This API is a backend for the TODO example
+  version: 0.1.0
+
+paths:
+  /todo-items:
+    get:
+      operationId: list-todo-items
+      description: Lists all TODO items
+      responses:
+        "200":
+          description: Ok
+          content:
+            "application/json":
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/todo-item"
+
+    post:
+      operationId: add-todo-item
+      description: Add a todo-item
+      requestBody:
+        required: true
+        content:
+          "application/json":
+            schema:
+              $ref: "#/components/schemas/todo-item"
+      responses:
+        "201":
+          description: Created
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/todo-item"
+
+  /todo-items/{id}:
+    put:
```

Comment:
I'm not too sure if we want a put.

So the things we want to change on a todo item, is
- we want to change the name
- we want to mark them complete

I'd model this as two seperate actions

Reply:
So for the change of name, a Patch request would be used instead to update the name. And an endpoint for "mark as done" will also be created and initially be set to false

Reply:
I understand what you're saying. I made change to the API spec!, they are specified to do what we want now. I used a PATCH request to handle the change of Todo description and mark as done. Please have a look at my latest commits so I can know if there's something I should change or we can move on! 🚀

Reply:
It's getting there, but I feel the oneOf solution is making things a bit more complicated than they have to be.

Of course this is a bit nit picky because this is just a small project, but the simplest solution is often the best solution. Things should be as simple as possible (but not more simple than that).

I took the liberty of changing a few things in a pr -> https://github.com/IsraelOjiefoh/SkiffaExamples/pull/2 <-.